core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1495,6 +1517,13 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
     distinctRoles
   }
 
+  def metadataLogDir: String = {
+    Option(getString(KafkaConfig.MetadataLogDirProp)) match {
+      case Some(dir) => dir
+      case None => logDirs.head
```

Comment:
Does it cause trouble if we change the order of directories from "log.dirs" when restarting server?

Reply:
Good question. I think the implementation will have to be smart enough to find the metadata log dir. Let me see if it is straightforward to add that here or if we should leave it for a follow-up. I guess the alternative would be to require `metadata.log.dir`, but I think we were trying to avoid that.

Reply:
What I am thinking for the moment is the following:

1. Assume a deterministic mapping to the metadata dir.
2. Raise an exception on startup if we find the `@metadata` log dir in another directory.